### ISSUE TITLE
Upgrading ArcGIS Desktop from 10.6.1 to 10.8.1 with DSC 3.1.0 is not working

### ISSUE THREAD

**Issue by eanderson2030** [Esri/arcgis-powershell-dsc#285]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request

### Module Version

* 3.1.0

### Affected Resource(s)

* ArcGIS_Desktop

### Configuration Files

```
{
    "AllNodes": [
        {
            "NodeName": "<FQDN>",
            "Role": [
                "Desktop"
            ]
        }
    ],
    "ConfigData": {
        "DesktopVersion": "10.8.1",
	"OldVersion": "10.6.1",
        "Desktop": {
            "SeatPreference": "Float",
            "InstallFeatures": "ALL",
            "ModifyFlexdAcl": "FALSE",
            "BlockAddIns": "0",
            "DesktopConfig": "false",
            "EsriLicenseHost": "<FQDN>",
            "SoftwareClass": "Professional",
            "EnableEUEI": "False",
            "Installer": {
                "Path": "\\\\....\\ArcGIS_Desktop_1081_175110.exe",
                "InstallDir": "E:\\Programs\\ArcGIS\\Desktop",
                "InstallDirPython": "E:\\Python27",
                "PatchesDir": "\\\\.....\\1081\\patches"
            }
        }
    }
}
```

### Expected Behavior

Upgrade ArcGIS Desktop from 10.6.1 to 10.8.1

### Actual Behavior

Nothing. Running from Powershell ISE as Admin and when the below command is passed, it simply drops to a new line. No error message or log file is written. 

### Steps to Reproduce

Run the below command from PowerShell ISE as Admin referencing the above .json file:

Invoke-ArcGISConfiguration -ConfigurationParametersFile "E:\...\configs\310\Desktop-Float.json" -Mode Upgrade

### Important Factoids

What we've tested:

- **Only reproducible with ArcGIS Desktop upgrade**. ArcGIS Server upgrade completes successfully. 

- Originally was running command from same machine we were attempting to upgrade ArcGIS Desktop on, but running from another machine also failed

- Tried using the above config to do a fresh install of ArcGIS Desktop 10.6.1 (successful) on a separate machine also using 3.1.0, and then used same config to perform an upgrade to 10.8.1 (but adding OldVersion parameter and updating install exe and patches pathway), but this gave same issue. Would just drop to a new line in PowerShell ISE - no error message or output logs

- Modified install path (removed a few characters) and the behavior was the same (no error or logs written), which tells me its not an issue with the config, but something with the module itself 

- Somewhat related question: Where can we expect output logs to be written to? Even when we did the fresh install, which succeeded, there were no log files written to the location set in PowerShell ISE.

**Comment by shailesh91** [comment]
@eanderson2030 Upgrades are supported only for Enterprise deployments. You can patch Desktop or pro deployments but not upgrade them. Thanks.

**Comment by eanderson2030** [comment]
Hi @shailesh91 , thanks for your response. The Wiki page didn't make that part super clear, but appreciate you clarifying.

**Comment by cameronkroeker** [comment]
@eanderson2030 The workflow for upgrading ArcGIS Desktop, ArcGIS Pro and License Manager are a bit different than upgrading ArcGIS Enterprise. For ArcGIS Desktop, Pro, and License Manager `-Mode Install` and `-Mode InstallLicense` can be used rather than `-Mode Upgrade`. We will be creating various Wiki pages that will outline the correct workflow.

In your case update `"DesktopVersion": "10.6.1"`, to `"DesktopVersion": "10.8.1"`, (no need for `"OldVersion": "10.6.1"`, since we will be using `-Mode Install`). Then update the installer path to point to the 10.8.1 setup (which it looks like you already have).

Then run ```Invoke-ArcGISConfiguration -ConfigurationParametersFile "E:...\configs\310\Desktop-Float.json" -Mode Install```

This will upgrade ArcGIS Desktop from 10.6.1 to 10.8.1. Also keep in mind looks like you are using a floating license (concurrent use) so you will want to first upgrade License Manager so that the ArcGIS Desktop can retrieve the updated license.

Thanks,
Cameron K.

**Comment by eanderson2030** [comment]
Thanks @cameronkroeker ! That worked for us.

**Comment by cameronkroeker** [comment]
> Thanks @cameronkroeker ! That worked for us.

Great! Glad to hear that worked for you.

**Comment by cameronkroeker** [comment]
Released v3.1.1 and added the following wiki pages:

> ##### [Upgrade an ArcGIS Pro Single Use deployment](V3.-Upgrade-an-ArcGIS-Pro-Single-Use-deployment)
> ##### [Upgrade an ArcGIS Desktop Single Use deployment](V3.-Upgrade-an-ArcGIS-Desktop-Single-Use-deployment)
> ##### [Upgrade an ArcGIS License Manager deployment](V3.-Upgrade-an-ArcGIS-License-Manager-deployment)
